realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -0,0 +1,604 @@
+package io.realm;
+
+import java.util.AbstractList;
+import java.util.ConcurrentModificationException;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.ListIterator;
+
+import io.realm.internal.Collection;
+import io.realm.internal.InvalidRow;
+import io.realm.internal.RealmObjectProxy;
+import io.realm.internal.SortDescriptor;
+import io.realm.internal.Table;
+import io.realm.internal.UncheckedRow;
+
+/**
+ * General implementation for {@link OrderedRealmCollection} which is based on the {@link Collection}.
+ * Currently only {@link RealmResults} and {@link OrderedRealmCollectionSnapshot} is inherited from this class. But
```

Comment:
extend? Since its subject is plural, `{@link RealmResults} and {@link OrderedRealmCollectionSnapshot}`.